pkgs/by-name/ol/ollama/package.nix

```diff
@@ -185,7 +185,8 @@ goBuild ((lib.optionalAttrs enableRocm {
     mv "$out/bin/ollama" "$out/bin/.ollama-unwrapped"
     makeWrapper "$out/bin/.ollama-unwrapped" "$out/bin/ollama" ${
       lib.optionalString enableRocm
-        ''--set-default HIP_PATH '${rocmPath}' ''} \
+        ''--set-default HIP_PATH '${rocmPath}' \
+          --set-default HSA_OVERRIDE_GFX_VERSION 0 ''} \
```

Comment:
What does 0 mean here?

Reply:
Ho good catch !

Reply:
It's essentially garbage. It could just as well be a "1" or "." or "-". I added it because of its interaction with rocblas compiled without tensile files.

See #321335 and #320202 for details, tldr rocblas compiled with tensile files (large lazy loaded gpu specific data (and/or libraries?)) takes a lot of processing, memory, and storage to compile, and hydra won't cache it because it's too large or something like that. Most ollama users are unable to compile it, their machines choking and even crashing. So, rocblas was recently changed back to not compile tensile files, which works fine for some gpus, but for others (including mine), rocm becomes unable to detect the gpu unless HSA_OVERRIDE_GFX_VERSION is set.

I very recently figured out that the actual value of HSA_OVERRIDE_GFX_VERSION doesn't actually matter for my gpu (6950 xt), and any nonempty string will enable rocm for gfx 1030, while an empty string or being unset will cause ollama to only use cpu. Thus the garbage value of "0" causes rocm to work correctly for me without any further configuration, and my hope is that this will also apply to other users, allowing them to use rocm without having to figure out some arcane override (it's not like I really care about my own setup working with ollama, I don't actually even use ollama).

Reply:
> rocm becomes unable to detect the gpu unless HSA_OVERRIDE_GFX_VERSION is set

Could you add this to a comment near it?